src/index.d.ts

```diff
@@ -627,3 +627,20 @@ export class LruCacheObject implements ICacheObject {
   remove (key: string|number): void;
   clear (): void;
 }
+
+export class FlatMapCacheObject implements ICacheObject {
+  set (key: any, selectorFn: any): void;
+  get (key: any): any;
+  remove (key: any): void;
+  clear (): void;
+  isCacheKeyValid (): boolean;
```

Comment:
Should take `cacheKey: any` as a parameter. Also, there's the `Resolver` definition in this typings that restricts its return value to `string | number`, should be `any` now.